pyfr/backends/base/kernels.py

```diff
@@ -116,6 +116,12 @@ def _build_arglst(self, dims, argn, argt, argdict):
                 else:
                     raise
 
+            # Check that argument is not a row sliced matrix
+            if (isinstance(ka, self.backend.matrix_slice_cls) and
```

Comment:
Move the slice to the end of mattypes so you can compare against mattypes[-1].  Then move this check into block of L126.  The exception should also be a ValueError (as the caller has passed something which is invalid per the spec).  Can also just say row sliced matrices are not supported.